src/components/url-form.tsx

```diff
@@ -0,0 +1,90 @@
+'use client';
+
+import { Form, FormControl, FormField, FormItem } from './ui/form';
+import { Input } from './ui/input';
+import { Button } from './ui/button';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { cn } from '@/lib/utils';
+import ErrorAlert from './error-alert';
+import { useState } from 'react';
+
+const formSchema = z.object({
+  url: z
+    .string()
+    .min(1, 'Field Required')
+    .regex(
+      /^(https:\/\/)(www\.)?github\.com\/[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+\/?$/,
+      // /^(https:\/\/)(www\.)?github\.com\/[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+.*\/?$/, // This regex includes the possibility of having a path & query params after the repo name
+      'Not a valid Github Repo URL'
+    ),
+});
```

Comment:
regex looks good but zod also comes with a built in .url() type, we should use that, something like min(..).url().regex()